packages/react/src/state.tsx

```diff
@@ -21,6 +26,11 @@ type IncrementStateModifier = {
   type: 'increment'
   value: number
 }
+type LookupStateModifier = {
```

Comment:
hmm, this one is interesting because it transforms the payload of the original event prop and passes it into state. different from the other state modifiers and maybe it should be `lookupPayload` or just `payload` with an optional lookup field on it

Reply:
I think `payload` with optional lookup field makes sense. I would name it `path`. And we can create `payload` a function!

Reply:
renamed to `eventPayload` as we already had this name. Maybe it's a bit lengthy but we can always change.